sycl/include/CL/sycl/detail/clusm.hpp

```diff
@@ -108,23 +105,24 @@ class CLUSM {
 } // namespace usm
 
 namespace cliext {
-bool initializeExtensions(cl_platform_id platform);
+bool initializeExtensions(cl_context context, cl_platform_id platform);
 } // namespace cliext
 
 } // namespace detail
 } // namespace sycl
 } // namespace cl
 
-__SYCL_EXPORTED extern cl::sycl::detail::usm::CLUSM *gCLUSM;
-inline cl::sycl::detail::usm::CLUSM *GetCLUSM() {
-  if (gCLUSM == nullptr) {
-    cl::sycl::detail::usm::CLUSM::Create(gCLUSM);
-  }
-
+__SYCL_EXPORTED extern std::map<cl_context, cl::sycl::detail::usm::CLUSM *>
+    gCLUSM;
+inline cl::sycl::detail::usm::CLUSM *GetCLUSM(cl_context ctxt) {
   cl::sycl::detail::usm::CLUSM *retVal = nullptr;
+  if (gCLUSM.find(ctxt) == gCLUSM.end()) {
```

Comment:
Suggest rewriting it to:

cl::sycl::detail::usm::CLUSM *&retVal  = gCLUSM[ctxt];
if(nullptr == retVal)
  retVal = new cl::sycl::detail::usm::CLUSM();

And move check for backend to the function beginning. Now you are allocating CLUSM even for non OpenCL BE.